src/config/config.go

```diff
@@ -47,35 +50,58 @@ func printHelpAndExit(err error) {
+	flagSet.BoolVar(&ret.SkipCertVerification, "insecure", false, "Do not validate TLS certificate")
+	flagSet.StringVar(&ret.CACertPath, "cacert", "", "Path to custom CA certificate in PEM format")
+	flagSet.StringVar(&ret.HTTPProxy, "http-proxy", "", "A proxy URL to use for creating HTTP connections outside the cluster")
+	flagSet.StringVar(&ret.HTTPSProxy, "https-proxy", "", "A proxy URL to use for creating HTTPS connections outside the cluster")
+	flagSet.StringVar(&ret.NoProxy, "no-proxy", "", "A comma-separated list of destination domain names, domains, IP addresses, or other network CIDRs to exclude proxying")
+	flagSet.StringVar(&ret.ServiceIPs, "service-ips", "", "All IPs of assisted service node")
+	flagSet.StringVar(&ret.HighAvailabilityMode, "high-availability-mode", "", "high-availability expectations, \"Full\" which represents the behavior in a \"normal\" cluster. Use 'None' for single-node deployment. Leave this value as \"\" for workers as we do not care about HA mode for workers.")
+	flagSet.BoolVar(&ret.CheckClusterVersion, "check-cluster-version", false, "Do not monitor CVO")
+	flagSet.StringVar(&ret.MustGatherImage, "must-gather-image", "", "Custom must-gather image")
+	flagSet.Var(&ret.DisksToFormat, "format-disk", "Disk to format. Can be specified multiple times")
+	flagSet.BoolVar(&ret.SkipInstallationDiskCleanup, "skip-installation-disk-cleanup", false, "Skip installation disk cleanup gives disk management to coreos-installer in case needed")
+
+	// Add dry-run specific flag bindings.
+	err := envconfig.Process("dryconfig", &DefaultDryRunConfig)
+	if err != nil {
+		fmt.Printf("envconfig error: %v", err)
+		os.Exit(1)
+	}
+
+	flagSet.BoolVar(&ret.DryRunEnabled, "dry-run", DefaultDryRunConfig.DryRunEnabled, "Dry run avoids/fakes certain actions while communicating with the service")
+	flagSet.StringVar(&ret.ForcedHostID, "force-id", DefaultDryRunConfig.ForcedHostID, "The fake host ID to give to the host")
+	flagSet.StringVar(&ret.FakeRebootMarkerPath, "fake-reboot-marker-path", DefaultDryRunConfig.FakeRebootMarkerPath, "A path whose existence indicates a fake reboot happened")
+	flagSet.StringVar(&ret.DryRunClusterHostsPath, "dry-run-cluster-hosts-path", DefaultDryRunConfig.DryRunClusterHostsPath, "A path to a JSON file with information about hosts in the cluster")
+
+	flagSet.Parse(os.Args[1:])
+
+	//Process dry-run arguments if necessary.
+	if ret.DryRunEnabled {
+		if parseErr := DryParseClusterHosts(ret.DryRunClusterHostsPath, &ret.ParsedClusterHosts); parseErr != nil {
+			fmt.Printf("Error parsing cluster hosts: %v", parseErr)
+			os.Exit(1)
+		}
+	}
 
 	var installerArgs string
-	flag.StringVar(&installerArgs, "installer-args", "", "JSON array of additional coreos-installer arguments")
-	h := flag.Bool("help", false, "Help message")
-	flag.Parse()
+	flagSet.StringVar(&installerArgs, "installer-args", "", "JSON array of additional coreos-installer arguments")
+	h := flagSet.Bool("help", false, "Help message")
```

Comment:
These need to be set up before the call to `flagSet.Parse`

Reply:
+1